future/transaction.go

```diff
@@ -621,31 +623,66 @@ func MakeApplicationCreateTx(
 	accounts []string,
 	foreignApps []uint64,
 	foreignAssets []uint64,
-	appBoxReferences []types.AppBoxReference,
 	sp types.SuggestedParams,
 	sender types.Address,
 	note []byte,
 	group types.Digest,
 	lease [32]byte,
 	rekeyTo types.Address) (tx types.Transaction, err error) {
-
-	oncomp := types.NoOpOC
-	if optIn {
-		oncomp = types.OptInOC
-	}
-
-	return MakeApplicationCallTx(
+	return MakeApplicationCreateTxWithBoxes(
+		optIn,
+		approvalProg,
+		clearProg,
+		globalSchema,
+		localSchema,
 		0,
 		appArgs,
 		accounts,
 		foreignApps,
 		foreignAssets,
-		appBoxReferences,
-		oncomp,
+		nil,
+		sp,
+		sender,
+		note,
+		group,
+		lease,
+		rekeyTo,
+	)
+}
+
+// MakeApplicationCreateTxWithExtraPages makes a transaction for creating an application (see above for args desc.)
+// - optIn: true for opting in on complete, false for no-op.
+//
+// NOTE: if you need to use boxes, use MakeApplicationUpdateTxWithBoxes instead.
```

Comment:
@jasonpaulos Minor:  Seems the comment's intent is to reference _create_ instead of _update_. Can you double check me?

```suggestion
// NOTE: if you need to use boxes, use MakeApplicationCreateTxWithBoxes instead.
```

Reply:
I think you are right, there is one other case need to be taken care of, and the rest is fine.

Reply:
Thanks for noticing, this was a copy-paste error. It's fixed in 9b8ad96a29b8b01fe3ddf129e0249f4f2b7f6e18